### DISCUSSION TITLE
Ability to call internal backend functions for user

### DISCUSSION THREAD

**Discussion by yohanchatelain** [verificarlo/verificarlo#270] [Q&A]
Hi,
I know that is a project that has been discussed for the interflop project but I would like to know if you plan to release it for verificarlo? It would be helpful for the fuzzy-libmath project because for the moment we use a workaround but having the `_mca_inexact_binary{32,64}` functions would be a cleaner way to add noise.

**Comment by pablooliveira** [comment]
Hi @yohanchatelain. Indeed, this feature has been discussed within `interflop` but we have not yet reached an agreement on how to implement it.

Therefore we do not have an implementation. What would you need / like in the context of `fuzzy-libmath` ?

**Reply by yohanchatelain** [reply]
Ok thanks. In `fuzzy-libmath` we perturb the output of the math function. To do so, we add `0` to the result to trigger the verificarlo instrumentation. The problem with this approach is that we can't use the RR mode for the virtual precision 24 for float and 53 for double functions. Indeed, the result of the math function will always be representable and no noise will be added. Having the `_mca_inexact_binary{32,64}` will allow us to directly perturb the output without using the +0 trick.
